pkg/operator/sync.go

```diff
@@ -633,6 +639,60 @@ func (optr *Operator) syncCustomResourceDefinitions() error {
 				return err
 			}
 		}
+		if !equality.Semantic.DeepEqual(c.Spec, current.Spec) || !equality.Semantic.DeepEqual(c.Status, current.Status) {
+			// we have changes. And need to regen the CR immediately or else we face validation errors.
+			// right now, the only changes we know of that we support is the controller certificate migration.
+			config, err := optr.client.MachineconfigurationV1().ControllerConfigs().Get(context.TODO(), "machine-config-controller", metav1.GetOptions{})
+			if err != nil {
+				if os.IsNotExist(err) {
+					break
+				}
+				return err
+			}
+			names := []string{
+				"KubeAPIServerServingCAData", "CloudProviderCAData", "RootCAData", "AdditionalTrustBundle",
+			}
+
+			certs := [][]byte{
```

Comment:
I'm a bit hesitant about hard coding it like this. Would it be possible to either:
1. detect the diff and temporarily shove in dummy data into the CR generically, or
2. do a diff detection + update mechanism here? (i.e. check if the diff contains any of these, if it does, do this, otherwise just return)